pkg/controller/certificates/signer/signer.go

```diff
@@ -203,6 +206,24 @@ func (s *signer) sign(x509cr *x509.CertificateRequest, usages []capi.KeyUsage, n
 	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: der}), nil
 }
 
+func (s *signer) duration(expirationSeconds *int32) time.Duration {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSRDuration) {
+		return s.certTTL
+	}
+
+	if expirationSeconds == nil {
+		return s.certTTL
+	}
+
+	// honor requested duration is if it is less than the default TTL
+	// use 10 min (2x hard coded backdate above) as a sanity check lower bound
+	if requestedDuration := csr.ExpirationSecondsToDuration(*expirationSeconds); requestedDuration < s.certTTL && requestedDuration >= 10*time.Minute {
```

Comment:
if this is less than our lower bound we ignore it, rather than clip to our lower bound?

Reply:
Let me split the two cases.  It does not matter today but it could in the future if we changed API validation.